Userland/Libraries/LibWeb/HTML/HTMLImageElement.cpp

```diff
@@ -302,6 +302,7 @@ WebIDL::ExceptionOr<JS::NonnullGCPtr<JS::Promise>> HTMLImageElement::decode() co
                 return false;
 
             auto exception = WebIDL::EncodingError::create(realm, "Current request state is broken"_fly_string);
+            HTML::TemporaryExecutionContext context(HTML::relevant_settings_object(*this));
```

Comment:
This feels like a structural problem with the entire method. There's another instance in `reject_if_document_not_fully_active` where we could be in the deferred_invoke without having an execution context.

I think we should slap a context at the top of the queue_microtask, and the top of the deferred_invoke on 322, instead of only having it in two places. (and the context should have the callback flag set).

Reply:
Unless it's guaranteed to have an execution context in queue_a_microtask? In which case it would only be needed in the deferred_invoke

Reply:
After staring at the code for a bit, I concluded that we (probably) need to ensure a temporary context exists in all three code paths.